lib/Dialect/Verif/VerifOps.cpp

```diff
@@ -82,6 +83,37 @@ static LogicalResult canonicalize(Op op, PatternRewriter &rewriter) {
 } // namespace AssertLikeOp
 
 LogicalResult AssertOp::canonicalize(AssertOp op, PatternRewriter &rewriter) {
+  // verif.assert %a if %true -> verif.assert %a
+  if (op.getEnable()) {
+    if (auto enableConst = op.getEnable().getDefiningOp<hw::ConstantOp>()) {
+      if (enableConst.getValue().isOne()) {
+        rewriter.modifyOpInPlace(op, [&]() { op.getEnableMutable().clear(); });
+        return success();
+      }
+    }
+  }
+
+  // verif.assert %a if %false -> <delete>
+  if (op.getEnable()) {
+    if (auto enableConst = op.getEnable().getDefiningOp<hw::ConstantOp>()) {
+      if (enableConst.getValue().isZero()) {
+        rewriter.eraseOp(op);
+        return success();
+      }
+    }
+  }
```

Comment:
Could you fuse these if-statements?

Reply:
This has been switched around in the later commits.